core/setup/kubernetes.go

```diff
@@ -1,40 +1,45 @@
 package setup
 
 import (
-//	"crypto/tls"
-//	"crypto/x509"
+//  "crypto/tls"
+//  "crypto/x509"
     "fmt"
-//	"io/ioutil"
-//	"net"
-//	"net/http"
-//	"time"
+//  "io/ioutil"
+//  "net"
+//  "net/http"
+    "strings"
+//  "time"
 
-	"github.com/miekg/coredns/middleware"
-	"github.com/miekg/coredns/middleware/kubernetes"
-	k8sc "github.com/miekg/coredns/middleware/kubernetes/k8sclient"
-	"github.com/miekg/coredns/middleware/proxy"
-//	"github.com/miekg/coredns/middleware/singleflight"
+    "github.com/miekg/coredns/middleware"
+    "github.com/miekg/coredns/middleware/kubernetes"
+    "github.com/miekg/coredns/middleware/kubernetes/nametemplate"
+    k8sc "github.com/miekg/coredns/middleware/kubernetes/k8sclient"
+    "github.com/miekg/coredns/middleware/proxy"
+//  "github.com/miekg/coredns/middleware/singleflight"
 
-	"golang.org/x/net/context"
+    "golang.org/x/net/context"
 )
 
-const defaultK8sEndpoint = "http://localhost:8080"
+const (
+    defaultK8sEndpoint = "http://localhost:8080"
+    defaultNameTemplate = "${service}.${namespace}.${zone}"
```

Comment:
The logging middleware has a (simple) template language as well. Maybe it's a good idea to re-use that here. That would mean dropping the `$`